bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/client/Client.java

```diff
@@ -0,0 +1,150 @@
+                throw new EntsoeResponseException("Request failed");
+            }
+            logger.trace("Response: {}", responseText);
+            return parseXmlResponse(responseText);
+        } catch (IOException e) {
+            if (e.getMessage().contains("Authentication challenge without WWW-Authenticate header")) {
+                throw new EntsoeConfigurationException("Authentication failed. Please check your security token");
+            }
+            throw new EntsoeResponseException(e);
+        } catch (ParserConfigurationException | SAXException e) {
+            throw new EntsoeResponseException(e);
+        }
+    }
+
+    private List<EntsoeTimeSerie> parseXmlResponse(String responseText)
+            throws ParserConfigurationException, SAXException, IOException, EntsoeResponseException {
+        List<EntsoeTimeSerie> responseList = new ArrayList<>();
+
+        DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+        Document document = documentBuilder.parse(new InputSource(new StringReader(responseText)));
+        document.getDocumentElement().normalize();
+
+        // Check for rejection
+        if (document.getDocumentElement().getNodeName().equals("Acknowledgement_MarketDocument")) {
+            NodeList reasonOfRejection = document.getElementsByTagName("Reason");
+            Node reasonNode = reasonOfRejection.item(0);
+            Element reasonElement = (Element) reasonNode;
+            String reasonCode = reasonElement.getElementsByTagName("code").item(0).getTextContent();
+            String reasonText = reasonElement.getElementsByTagName("text").item(0).getTextContent();
+            throw new EntsoeResponseException(
+                    String.format("Request failed with API response: Code %s, Text %s", reasonCode, reasonText));
+        }
+
+        // Get all "timeSeries" nodes from document
+        NodeList listOfTimeSeries = document.getElementsByTagName("TimeSeries");
+        for (int i = 0; i < listOfTimeSeries.getLength(); i++) {
+            Node timeSeriesNode = listOfTimeSeries.item(i);
+            if (timeSeriesNode.getNodeType() == Node.ELEMENT_NODE) {
+                Element timeSeriesElement = (Element) timeSeriesNode;
+
+                String currency = timeSeriesElement.getElementsByTagName("currency_Unit.name").item(0).getTextContent();
+                String measureUnit = timeSeriesElement.getElementsByTagName("price_Measure_Unit.name").item(0)
+                        .getTextContent();
+
+                NodeList listOfTimeInterval = timeSeriesElement.getElementsByTagName("timeInterval");
+                Node startTimeNode = listOfTimeInterval.item(0);
+                Element startTimeElement = (Element) startTimeNode;
+                String startTime = startTimeElement.getElementsByTagName("start").item(0).getTextContent();
+                ZonedDateTime zonedStartTime = ZonedDateTime.parse(startTime);
```

Comment:
Since this is a UTC timestamp, and with the newly introduced class `EntsoeTimeSerie`, you could now consider `Instant` everywhere except `EntsoeTimeSerie.getTime` for simplicity. In case you want to follow this road, I'll make individual refactoring suggestions.
```suggestion
                Instant startTimeInstant = Instant.parse(startTime);
```